Interface for selecting logging options

rishabh-ranjan1 opened the pull request:
May I request you to have a look at my branch for adding interface for selecting logging options?
Please suggest any changes you would like me to make.
This has reference to my earlier message at
https://github.com/invesalius/invesalius3/discussions/702

Rishabh

vighneshbhat9945 commented:
@rishabh-ranjan1  please  can you guide me to set up this project locally.

paulojamorim commented:
Hi @rishabh-ranjan1 

I tested the pull request, and it's working. However, as it's only the graphical interface, I can't merge it because the tool is still incomplete. To accept the pull request, the following improvements are necessary:

- Use the wxFileDialog component to specify the log file's name.
- Set default log file in temp folder.
- Use the logging library (https://docs.python.org/3/library/logging.html) to capture at least 1 event of each level and save in the file.
- Add a TextCtrl component to display the content of the last log. Format the font in red color for Critical and Error levels.

You're on the right track. Good luck!

rishabh-ranjan1 commented:
Dear Sir
Thanks for testing out my code. So far I have added wxFileDialog as per
your instruction. I am working on implementing the logging code. Can you
suggest the right folder to add logging components of the code.
Regards
Rishabh

On Sat, Feb 10, 2024 at 8:02 PM Paulo Henrique Junqueira Amorim <
***@***.***> wrote:

> Hi @rishabh-ranjan1 <https://github.com/rishabh-ranjan1>
>
> I tested the pull request, and it's working. However, as it's only the
> graphical interface, I can't merge it because the tool is still incomplete.
> To accept the pull request, the following improvements are necessary:
>
>    - Use the wxFileDialog component to specify the log file's name.
>    - Set default log file in temp folder.
>    - Use the logging library (
>    https://docs.python.org/3/library/logging.html) to capture at least 1
>    event of each level and save in the file.
>    - Add a TextCtrl component to display the content of the last log.
>    Format the font in red color for Critical and Error levels.
>
> You're on the right track. Good luck!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/invesalius/invesalius3/pull/704#issuecomment-1937021396>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXJ7JIGEDIR4S2JGEGLQWX3YS6AJTAVCNFSM6AAAAABCY5CGBKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMZXGAZDCMZZGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

paulojamorim commented:
Hi @rishabh-ranjan1 

It seems to have improved! I noticed that you fixed an error of overwriting the configuration file when opening InVesalius.

About your doubt, you can create a log.py module in invesalius/data and add things related to recording and managing the log there.

A note about the interface. For some reason the preferences window is too big (at least in Ubuntu 20.04 with GNOME Flashback). Make it smaller when you can. 

See picture:

![screen](https://github.com/invesalius/invesalius3/assets/808364/6d31dc10-a920-461c-acb2-174a306ef4fe)

Regards,
Paulo Amorim

rishabh-ranjan1 commented:
Dear Sir
I have added a file named log.py in the Invesalius/data folder in my branch.
I have added 3 functions:
              a) configureLogging - used to start or reconfigure logging
              b) closeLogging - to close logging operation
              c) function_call_tracking_decorator - a decorator function to
be added to any function for tracking its call
I would appreciate your input, and also would like your suggestions for the
following:
1) Shall I make it into a class?
2) Directions about how to add logging to Invesalius functions

Regards
Rishabh Ranjan

On Mon, Feb 19, 2024 at 7:52 AM Paulo Henrique Junqueira Amorim <
***@***.***> wrote:

> Hi @rishabh-ranjan1 <https://github.com/rishabh-ranjan1>
>
> It seems to have improved! I noticed that you fixed an error of
> overwriting the configuration file when opening InVesalius.
>
> About your doubt, you can create a log.py module in invesalius/data and
> add things related to recording and managing the log there.
>
> A note about the interface. For some reason the preferences window is too
> big (at least in Ubuntu 20.04 with GNOME Flashback). Make it smaller when
> you can.
>
> See picture:
>
> screen.png (view on web)
> <https://github.com/invesalius/invesalius3/assets/808364/6d31dc10-a920-461c-acb2-174a306ef4fe>
>
> Regards,
> Paulo Amorim
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/invesalius/invesalius3/pull/704#issuecomment-1951583430>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXJ7JIG23YMPHN5MIFCFCT3YUKZNTAVCNFSM6AAAAABCY5CGBKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNJRGU4DGNBTGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

paulojamorim commented:
Hi @rishabh-ranjan1 

Sorry for late reply. About your questions.

1) Yes, it's interesting to make a class out of it. Keep it as a singleton class.

2) Change InVesalius prints for logging.

In console logging window, do like this in wxPython to show: https://stackoverflow.com/questions/73440649/how-to-debug-log-messages-on-a-window-from-different-modules/73440890#73440890

Regards,
Paulo Amorim
